modules/seastate/src/SeaState_C_Binding.f90

```diff
@@ -289,19 +291,32 @@ SUBROUTINE SeaSt_C_End(ErrStat_C,ErrMsg_C) BIND (C, NAME='SeaSt_C_End')
     CALL SeaSt_End(u, p, x, xd, z, OtherState, y, m, ErrStat2, ErrMsg2)
 
     CALL SetErrStat( ErrStat2, ErrMsg2, ErrStat, ErrMsg, RoutineName )
-    CALL SetErr( ErrStat, ErrMsg, ErrStat_C, ErrMsg_C )
+    CALL SetErrStat_F2C( ErrStat, ErrMsg, ErrStat_C, ErrMsg_C )
 
 END SUBROUTINE
 
-! SUBROUTINE get_wave_height(position)
-
-
-! SUBROUTINE get_wave_field_pointer()
-! pass back the internal pointer to the wave field to the calling code
-! END SUBROUTINE
-
-! SUBROUTINE set_flow_field_pointer()
+FUNCTION SeaSt_GetWaveFieldPointer_C() BIND (C, NAME='SeaSt_GetWaveFieldPointer_C')
+IMPLICIT NONE
+#ifndef IMPLICIT_DLLEXPORT
+!DEC$ ATTRIBUTES DLLEXPORT :: SeaSt_GetWaveFieldPointer_C
+!GCC$ ATTRIBUTES DLLEXPORT :: SeaSt_GetWaveFieldPointer_C
+#endif
+    TYPE(C_PTR) :: SeaSt_GetWaveFieldPointer_C
+    SeaSt_GetWaveFieldPointer_C = C_LOC(p%WaveField)
+    RETURN
+END FUNCTION
+
+! FUNCTION SetFlowFieldPointer_C(FlowFieldPointer) BIND (C, NAME='SetFlowFieldPointer_C')
```

Comment:
I'm not sure three is a need for a `SetFlowFieldPointer` here.

Reply:
Resolved